doc/classes/NavigationAgent2D.xml

```diff
@@ -0,0 +1,159 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="NavigationAgent2D" inherits="Node" version="3.4">
+	<brief_description>
+		2D Agent used in navigation for collision avoidance.
+	</brief_description>
+	<description>
+		2D Agent that is used in navigation to reach a location while avoiding static and dynamic obstacles. The dynamic obstacles are avoided using RVO collision avoidance. The agent needs navigation data to work correctly. This can be done by having the agent as a child of a [Navigation2D] node, or using [method set_navigation]. [NavigationAgent2D] is physics safe.
```

Comment:
```suggestion
		2D agent used in navigation for collision avoidance.
	</brief_description>
	<description>
		2D agent that is used in navigation to reach a location while avoiding static and dynamic obstacles. The dynamic obstacles are avoided using RVO (Reciprocal Velocity Obstacles) collision avoidance. The agent needs navigation data to work correctly. This can be done by having the agent as a child of a [Navigation2D] node, or using [method set_navigation]. [NavigationAgent2D] is physics safe.
```

Reply:
Same comments as for the 3D counterpart for the rest of the docs. (Good docs btw!)